### ISSUE TITLE
arrow functions on separate line can be confusing

### ISSUE THREAD

**Issue by hyperknot** [prettier/prettier#4303]
**Prettier 1.11.1**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEEBucBOAbAhgTwGUYJM4A6AZwAsIB3AFTgA8YAKAHSgAJuOQAopkykk3AHIQY3bBADmcuABNuASyj8ANHx58QABWxxclONzKzcKmNTMAHXIu64oK2XLVRyWnbzYBKbgBeAD5uMFxsbAAjXDAAazYoAFco7TJKcmiIJXxyLBFMfy5-AG4uLnQsPCISMipaRhZ2Ll5+IUKxSWlcZJtSVQAvZR9WvUNjU3M4S2tbbgcnFzd5T246VRtuGzNMVTlqaWTTTG8QbTGA4LCIqNiEpNTsdLhM7Nz84VJiqDKuc9QdhgqmglGQoFwXzo+khCDBKFwaAgqiUAJMMGQADNIqZNCBopg4vE4DBCA4wOo5MgYJhknA8UoIGAsTj6SB1CcYPpCXIALa4FnYXEgABWlGYACFCQkSYRcLy4AAZdRwQXCiB9Ox9ABMarZDkwJ2Q+Nw0XwsigALse1gAHUUTZkABGAAMLrx1ogplthLsxutrywGABZAAjslVGRuY5+Xq8aZeao4yBKJSjABFZJSVVIGl0vEwU32pSOpDaguE1TYSkAYQgvNjKFeAFYAcc4AxTfDsUK4ABfPtAA)
```sh
--print-width 100
--no-semi
--single-quote
--trailing-comma es5
```

**Input:**
```jsx
overlayStore.showText(
  "Error: Not logged in", 
  "Please reload the page and log in.", 
  () => callback(null, res.body.error)
);

overlayStore.showText(
  "Error: Not authorized", 
  "Please reload the page and log in with the right user.", 
  () => callback(null, res.body.error)
);

```

**Output:**
```jsx
overlayStore.showText('Error: Not logged in', 'Please reload the page and log in.', () =>
  callback(null, res.body.error)
)

overlayStore.showText(
  'Error: Not authorized',
  'Please reload the page and log in with the right user.',
  () => callback(null, res.body.error)
)

```

**Expected behavior:**
Short version: I'd really like to keep those 3 parameters in 3 separate lines. At least to make sure that I'm not making a new line in the middle of a parameter.

Having the arrow of an arrow function on separate line can be quite confusing. In this example, I'd really like to have both functions be indented the same way, to make sure that I can visually check that all of them are in an arrow function and not called directly. Right now, prettier makes such organisation impossible.

**Comment by hyperknot** [comment]
What seems buggy, is that if I remove the `() => ` part from the first example, the code actually gets **shorter**, yet prettier prints it in **separate lines**, nicely.
